src/lib/util.js

```diff
@@ -1005,6 +1009,83 @@ function getInterfaceTreeInfo(iface, interfaces) {
 }
 
 
+/**
+ * Return a schema validation function for the given schema name.
+ *
+ * The function is provided by the Ajv library. The schema name can be one of:
+ * "extract-xxx.json" to target schemas under browserlib, "css", "dfns", ...
+ * to target schemas under files/extracts, "index.json", "events.json",
+ * "idlnames.json", "idlparsed" or "idlnamesparsed". Additional schemas may be
+ * added over time as more extraction facilities are added to Reffy.
+ *
+ * @function
+ * @public
+ * @param {any} data The data to validate
+ * @param {String} schemaName The name of the JSON schema to use
+ * @return {function} The "validate" function for Ajv. The function returns null
+ *   if the requested schema does not exist.
+ */
+function getSchemaValidationFunction(schemaName) {
+    const schemasFolder = path.join(__dirname, '..', '..', 'schemas');
+    const schemaFile =
+        (schemaName === 'index.json') ? path.join('files', schemaName) :
+        (schemaName === 'idlnamesparsed') ? path.join('postprocessing', 'idlnames-parsed.json') :
+        (schemaName === 'idlparsed') ? path.join('postprocessing', 'idlparsed.json') :
+        schemaName.startsWith('extract-') ? path.join('browserlib', `${schemaName}.json`) :
+        schemaName.endsWith('.json') ? path.join('postprocessing', schemaName) :
+        path.join('files', 'extracts', `${schemaName}.json`);
+    let schema;
+    try {
+        schema = require(path.join(schemasFolder, schemaFile));
+    }
+    catch (err) {
```

Comment:
It threw in the first version I had, but then it seemed a bit rude to throw when the caller requested an unknown schema. Typically, returning `null` allows to avoid a `try... catch` in Webref:
https://github.com/w3c/webref/pull/731/files#diff-128accda0c348bbd52246995174cc7409be8cf4308f3a1389b5d9e47a556cb55R15-R18

Reply:
why it matters - to avoid a situation where you think you're all good because you didn't get a complaint when in fact you made a type in your schema name.

how - I assume we know the patterns and the names that can be applied to it - anything that doesn't match that should throw (but my assumption may very well be wrong :)

Reply:
I was trying not to hardcode the list of names so that we don't have to update that piece of code when we add more extraction tools and post-processing modules. The code can tell which names are correct for extraction modules by looking at `property` in `src/browserlib/reffy.json`. There is no automatic way to tell what files post-processing modules that operate at the crawl level may generate though.

I'll take the liberty to merge as-is. If you think that deserves an update, the floor is yours :)